.pre-commit-hooks.yaml

```
@@ -0,0 +1,9 @@
+---
+- id: generate-mocks
+  name: generate mocks
+  description: Generate mocks with mockery
+  language: golang
+  additional_dependencies:
+    - github.com/vektra/mockery/v2@latest
+  entry: mockery
```

Rather than using additional_dependencies, the pre-commit-hook will already have access to the file system.  We should just be able to use `go run .` to enact the `mockery ...` command I believe.

Also, we may only want this hook to run when `.go` files are in the commit.  What do you think?

```suggestion
  entry: go run .
  files: \.go$
```

---

I'm unsure it will launch the code you expect to run.

I will make some tests.

But, I think keeping mockery call here would be clearer.

---

Using additional dependencies like this won't allow people to specify version in pre commit file because no matter what version they reference, you're pulling latest

---

Indeed, I'm unsure if there is something that can be done here, I will have to check

---

For now,it seems to me, that it would imply to maintain the version in sync in this file vs the tag. Quite a pain

---

Is it possible let the users assume the responsibility for installing mockery? If that's the case this should suffice
```yaml
---
- id: generate-mocks
  name: generate mocks
  description: Generate mocks with mockery
  language: golang
  entry: mockery
  pass_filenames: false
```

---

Yes, of course, it makes sense.

But then having 2 rules could be a something to consider

---

I think your approach is the most commonly used in fact

https://github.com/search?q=path%3A%22%2F.pre-commit-hooks.yaml%22+golang&type=code